### PR TITLE
clients: properly detect and convert `ErrRecordMismatch`

### PR DESCRIPTION
**Pull request by unknwon**
https://github.com/sourcegraph/sourcegraph-accounts-sdk-go/pull/4 only gets the half-job done, which did not actually detect and convert error to `ErrRecordMismatch` 💩 

## Test plan

CI

### CODE REVIEW

File: clientv1_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+				return connect.NewError(connect.CodeNotFound, nil)
+			},
+			wantErr: ErrNotFound.Error(),
+		},
+		{
+			name: "record mismatch",
+			err: func() error {
+				detail, err := connect.NewErrorDetail(&clientsv1.ErrorRecordMismatch{})
+				require.NoError(t, err)
+				grpcErr := connect.NewError(connect.CodeUnknown, nil)
```
Comment (bobheadxi) [inline clientv1_test.go:40]: Should we use the precondition failed code in these tests for presumably more realistic response?

**Review by bobheadxi — APPROVED** [review]

File: clients/v1/clients.proto [hunk 2]
```diff
@@ -87,3 +87,5 @@ message SignOutSessionRequest {
 }
 
 message SignOutSessionResponse {}
+
+message ErrorRecordMismatch{}
```
Comment (eseliger) [inline clients/v1/clients.proto:91]: :P 

```suggestion
message ErrorRecordMismatch {}
```

Comment (eseliger) [reply clients/v1/clients.proto:91]: Docstrings would be helpful, both on what this message means, and where it's returned!

**Review by eseliger — APPROVED** [review]

File: clientv1_test.go (hunk 1 above)
Comment (unknwon) [reply clientv1_test.go:40]: I am kinda debating hard in my head, on the one hand, precondition failed code is right for this particular error, but on the other hand it makes this generic error converter unnecessarily(?) complicated to digest. 🤔 

On the server-side tho, does return precondition failed code (in another PR)

**Comment by unknwon** [comment]
Merging as-is, happy to address post-merge feedback!

**Pull request merged**